checks.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+
+# As described in `CONTRIBUTING.md`, this script checks Optuna's source codes for formatting,
+# coding style and type hints. The `-u` option enables automatic formatting by `black` and `isort`.
+
+git update-index --chmod=+x
+
+res_pip_list=$(pip freeze)
+missing_dependencies=()
+if [ ! "$(echo $res_pip_list | grep black)" ] ; then
+  missing_dependencies+=(black)
+fi
+if [ ! "$(echo $res_pip_list | grep flake8)" ] ; then
+  missing_dependencies+=(flake8)
+fi
+if [ ! "$(echo $res_pip_list | grep isort)" ] ; then
+  missing_dependencies+=(isort)
+fi
+if [ ! "$(echo $res_pip_list | grep mypy)" ] ; then
+  missing_dependencies+=(mypy)
+fi
+if [ ! ${#missing_dependencies[@]} -eq 0 ]; then
+  echo "The following dependencies are missed:" "${missing_dependencies[@]}"
+  read -p "Would you like to install the missing dependencies? (y/N): " yn
+  case "$yn" in [yY]*) ;; *) echo "abort." ; exit ;; esac
+  pip install "${missing_dependencies[@]}"
```

Comment:
`flake8` is indirectly installed via `hacking` so if `flake8` is missing, we actually need to install `hacking`.

Reply:
Thanks for the review! So, should we install not `flake8` but `hacking`, right?